Rebase on master branch

brian-kephart opened the pull request:
The master branch has diverged from this one quite a bit, so I rebased on master and resolved the conflicts. Only the two most recent commits are new.

New commits:
- Extract `yarn lint` command from CI so it can be run it before pushing without having to remember the options.
- Remove `version` from package.json since we're not publishing an NPM package.
- Add a `packageManager` field to prevent Corepack from using yarn v3 and adding files to the repo.

The merge conflicts were minor except for nav_menu.js. @texpert I would appreciate it if you'd take a look at that file to make sure it is as it should be.

texpert commented:
`nav_menu.js` looks good to me.

I don't quite get it about the change from `yarn eslint . --ext .js,.jsx,.ts,.tsx` to `yarn lint` - I am not very accustomed to JS ecosystem. Can you elaborate, please?!

Also, I'm in doubt about merging this PR into `owen2345:feature/version-3`. I'm afraid it could cause a mess. The correct way, IMO, is to rebase the `owen2345:feature/version-3` branch on `master`.

brian-kephart commented:
>I don't quite get it about the change from `yarn eslint . --ext .js,.jsx,.ts,.tsx` to `yarn lint`. Can you elaborate, please?!

I needed to run eslint locally. I couldn't remember`yarn eslint . --ext .js,.jsx,.ts,.tsx`, so I put that command in package.json as a script so we can just type `yarn lint` instead. I picked up that pattern from the Rails bundling gems, which use the same process to create `yarn build` and `yarn build:css`. If you look at the `scripts` key in package.json you will see that it's running the same command as before.

I would normally put a change like that in a different PR. In this case, though, I couldn't run the linter locally at all without adding `packageManager` to package.json to get Corepack to behave, so CI was failing every time I pushed. Since I was already changing that file, I went ahead and added the linter script as well.

> The correct way, IMO, is to rebase the owen2345:feature/version-3 branch on master.

That's what this PR does. I don't like it either.

In this case the branches have enough conflicts that rebasing is not much cleaner than a merge. I tried it both ways, and this PR is the version where I rebased, but I wasn't happy with the PR diff on either version.

I would gladly trash this entire PR for something cleaner if it's available. I suppose I could force-push the branch to rewrite the history without a PR, but I wasn't about to do that without getting eyes on the result first.

Right now though, I can't develop on this branch because there are too many changes on master that haven't been merged. I'm also worried that these branches will continue to diverge, creating more of these messes.

texpert commented:
@brian-kephart, thanks so much for explanation about adding the `yarn lint` command!

I have successfully rebased the `feature/version-3` branch on `master`, so you can merge your last 2 PRs into.

I'd propose after this to merge `feature/version-3` into `master`, renaming the PR to a 2.8 release, and to craft the 2.8 gem. Because it seems we have more things to do for a major release, and it is better to not gather huge PRs like this, but release more frequently smaller changes.
